### PR TITLE
CloudMonitor: Improve detail of MQL series labels

### LINKED ISSUE #58317: Google Cloud Monitoring: Default labels in MQL query
**What happened**:

When running a basic MQL query like:

```
fetch gce_instance::compute.googleapis.com/instance/cpu/utilization
| filter zone =~ 'us-central.*'
```

The time series labels returned displays a wrong value:

Screenshot from 2022-11-04 13-28-47

**What you expected to happen**:

The series to automatically select a meaningfull label, like in the builder.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.2.3
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by aangelisc**
This returns labels for MQL queries that provide more detailed information about the series in question, bringing the results closer to what would be attained from the query builder.

Any `metric` label value will be appended to the end of the series name. Unfortunately, the metric type is not returned as a part of the `timeSeriesQuery` response. This could be determined from the query itself but this may get difficult with more complex queries so I've left this out for now. 

I may attempt a refactor in future to bring the two label extraction methods more in line with eachother (or combine them entirely).

Fixes #58317

### CODE REVIEW

**Review by asimpson — APPROVED** [review]
:rocket: Nice work

**Review by andresmgot — COMMENTED** [review]
This is not returning a meaningful metric name with other queries:

Screenshot from 2022-12-05 09-48-06

Taking a look to the `extractTimeSeriesLabels` method for `filter` queries, it seems that what we also want to use labels that start with `resource.label`.

What I would do:

 - Do not use the `PointDescriptor` key value as the default metric name (since that's not different per time series).
 - Make `extractTimeSeriesDataLabels` return the `defaultMetricName` as `extractTimeSeriesLabels` does (that way we don't calculate the metricName per descriptor).
 - Use the same criteria to return the metric name than for filter series. Since we don't have the metric type in the response, use all `metric.label.*` and `resource.label.*`.
 - Bonus points if you extend the `pointIterator` interface to return those label values for both `timeSeries` and `timeSeriesData` so you can reuse the same function for both query types. If you find this very cumbersome you can skip it.

The result should be similar to what you get in the gcloud console:

Screenshot from 2022-12-05 10-20-55

**Comment by aangelisc** [comment]
>  Taking a look to the extractTimeSeriesLabels method for filter queries, it seems that what we also want to use labels that start with resource.label.

Good catch, I missed this 👍 

I'll look into the bigger refactor you've suggested, great feedback 😊

File: pkg/tsdb/cloudmonitoring/types.go [hunk 1]
```diff
@@ -250,6 +284,56 @@ func (ts timeSeries) valueType() string {
 	return ts.ValueType
 }
 
+func (ts timeSeries) getLabels(labelDescriptors []LabelDescriptor, groupBys []string) (data.Labels, string) {
```
Comment (andresmgot) [inline pkg/tsdb/cloudmonitoring/types.go:287]: same here, you don't need the labelDescriptors

**Review by andresmgot — APPROVED** [review]
LGTM, the label parsing logic is different enough so it's okay to have two different methods (even though some logic is the same). I just added some minor comments.

**Pull request merged**